lib/server.js

```diff
@@ -37,31 +37,57 @@ module.exports = class AlgoServer extends EventEmitter {
     } = args
 
     this.port = port
-    this.host = new AOHost({
-      aos, apiKey, apiSecret, agent, wsURL, restURL
-    })
+    // If provided with order definitions then create the algoHost instance
+    // Mostly just for backwards compatibility
+    if (aos) {
+      host = new AOHost({
+        aos, apiKey, apiSecret, agent, wsURL, restURL
+      })
+      this.setAlgoHost(host)
+    }
+
+    this.hbi = startHB(this)
+    this.wss = startWSS(this)
+  }
+
+  close () {
+    this.host.close()
```

Comment:
A check if the host exists is now required here

Reply:
good catch